File: frontend_flutter/lib/models/task.dart
class Task {
  final String id;
  final String icon;
  final String title;
  final String subtitle;
  final String time;
  final String timeColor; // 'teal' | 'orange' | 'red'
  final bool done;
  final String category;
  final String? reasoning; // "Why this?" — clinical reasoning (FR-160)

  const Task({
    required this.id,
    required this.icon,
    required this.title,
    required this.subtitle,
    required this.time,
    this.timeColor = 'teal',
    this.done = false,
    required this.category,
    this.reasoning,
  });

  factory Task.fromJson(Map<String, dynamic> j) => Task(
        id: j['id'].toString(),
        icon: (j['icon'] as String?) ?? '•',
        title: j['title'] as String,
        subtitle: (j['subtitle'] as String?) ?? '',
        time: (j['scheduled_time'] as String?) ?? '',
        timeColor: (j['time_color'] as String?) ?? 'teal',
        done: (j['is_done'] as bool?) ?? false,
        category: (j['category'] as String?) ?? '',
        reasoning: j['reasoning'] as String?,
      );

  Task copyWith({bool? done}) => Task(
        id: id,
        icon: icon,
        title: title,
        subtitle: subtitle,
        time: time,
        timeColor: timeColor,
        done: done ?? this.done,
        category: category,
        reasoning: reasoning,
      );
}

File: backend/app/schemas/task.py
from pydantic import BaseModel


class TaskOut(BaseModel):
    id: str
    title: str
    subtitle: str | None = None
    icon: str | None = None
    category: str | None = None
    scheduled_time: str | None = None
    time_color: str = "teal"
    phase: str | None = None
    task_date: str | None = None
    is_done: bool = False
    priority: int = 0

    @classmethod
    def from_row(cls, t) -> "TaskOut":
        return cls(
            id=str(t.id),
            title=t.title,
            subtitle=t.subtitle,
            icon=t.icon,
            category=t.category.value if t.category else None,
            scheduled_time=t.scheduled_time,
            time_color=t.time_color or "teal",
            phase=t.phase,
            task_date=t.task_date,
            is_done=t.is_done,
            priority=t.priority or 0,
        )


class TaskToggle(BaseModel):
    is_done: bool
